### ISSUE TITLE
pg_restore crashes DB

### ISSUE THREAD

**Issue by timmy59100** [zalando/postgres-operator#1312]
Please, answer some short questions which should help us to understand your problem / question better?

- **Which image of the operator are you using?** e.g. registry.opensource.zalan.do/acid/postgres-operator:v1.6.0
- **Where do you run it - cloud or metal? Kubernetes
- **Are you running Postgres Operator in production?** [yes ]
- **Type of issue?** [Bug report]

PG-Restore results in DB crash and fails.
On a normal bitnami docker image the restore is successful.

021-01-14 13:47:38,211 INFO: no action.  i am the leader with the lock
2021-01-14 13:47:38.399 UTC user= db= pid=66 LOG:  checkpoint complete: wrote 28730 buffers (37.4%); 0 transaction log file(s) added, 0 removed, 12 recycled; write=119.733 s, sync=1.154 s, total=121.578 s; sync files=181, longest=1.067 s, average=0.006 s; distance=425984 kB, estimate=425984 kB
2021-01-14 13:47:43.213 28 LOG {ticks: 0, maint: 0, retry: 0}
2021-01-14 13:47:45.907 UTC user= db= pid=60 LOG:  server process (PID 361) was terminated by signal 9: Killed
2021-01-14 13:47:45.907 UTC user= db= pid=60 DETAIL:  Failed process was running: COPY public.datastore (id, length, last_modified, data) FROM stdin;
2021-01-14 13:47:45.907 UTC user= db= pid=60 LOG:  terminating any other active server processes
2021-01-14 13:47:45.910 UTC user= db= pid=60 LOG:  archiver process (PID 91) exited with exit code 2
2021-01-14 13:47:45.942 UTC user= db= pid=60 LOG:  all server processes terminated; reinitializing
2021-01-14 13:47:46.243 UTC user= db= pid=506 LOG:  database system was interrupted; last known up at 2021-01-14 13:47:38 UTC
2021-01-14 13:47:47.243 UTC user= db= pid=506 LOG:  database system was not properly shut down; automatic recovery in progress
2021-01-14 13:47:47.261 UTC user= db= pid=506 LOG:  redo starts at 4/5F000300
2021-01-14 13:47:48,152 INFO: Lock owner: test-cluster-0; I am test-cluster-0
2021-01-14 13:47:48,364 INFO: establishing a new patroni connection to the postgres cluster
2021-01-14 13:47:48.367 UTC user=[unknown] db=[unknown] pid=507 LOG:  connection received: host=[local]
2021-01-14 13:47:48.367 UTC user=postgres db=postgres pid=507 FATAL:  the database system is in recovery mode
2021-01-14 13:47:48,368 WARNING: Retry got exception: 'connection problems'
2021-01-14 13:47:48.392 UTC user=[unknown] db=[unknown] pid=509 LOG:  connection received: host=[local]
2021-01-14 13:47:48.392 UTC user=postgres db=postgres pid=509 FATAL:  the database system is in recovery mode
/var/run/postgresql:5432 - rejecting connections
2021-01-14 13:47:48,493 INFO: Error communicating with PostgreSQL. Will try again later
2021-01-14 13:47:53.514 UTC user= db= pid=506 LOG:  invalid record length at 4/6B2900F0
2021-01-14 13:47:53.514 UTC user= db= pid=506 LOG:  redo done at 4/6B2900B8
2021-01-14 13:47:53.540 UTC user= db= pid=506 LOG:  checkpoint starting: end-of-recovery immediate
2021-01-14 13:47:56.208 UTC user= db= pid=506 LOG:  checkpoint complete: wrote 24149 buffers (31.4%); 0 transaction log file(s) added, 13 removed, 15 recycled; write=0.319 s, sync=2.184 s, total=2.678 s; sync files=5, longest=2.127 s, average=0.436 s; distance=199231 kB, estimate=199231 kB
2021-01-14 13:47:56.236 UTC user= db= pid=506 LOG:  MultiXact member wraparound protections are now enabled
2021-01-14 13:47:56.261 UTC user= db= pid=513 LOG:  autovacuum launcher started
2021-01-14 13:47:56.263 UTC user= db= pid=60 LOG:  database system is ready to accept connections
2021-01-14 13:47:56.448 UTC user= db= pid=516 LOG:  pg_cron scheduler started
2021-01-14 13:47:58.173 UTC user=[unknown] db=[unknown] pid=521 LOG:  connection received: host=[local]

**Comment by Jan-M** [comment]
I don't think this is a Postgres operator issue or Spilo issue. Check Postgres logs, observe pod metrics (memory in particular) on what is happening during your potentially large import using `COPY`
